internal/DebounceDecorator/tests/DebounceDecorator-specs.js

```diff
@@ -1,101 +0,0 @@
-import {mount} from 'enzyme';
-import DebounceDecorator from '../';
-
-describe('DebounceDecorator', () => {
-	test(
-		'should emit the event after the delay',
```

Comment:
I was able to make a test for `DebounceDecorator` after some searching. :)
Well, there are two things we should have in mind to write this kind of test.
First, we need to fire custom events, in this case, `onChange` and `onCancel`. For firing custom events with React Testing Library, we have `createEvent` method.
Secondly, we need to use `useRef` to attach the handler for these custom events to dispatch. Unlike the DOM events, we should use `addEventListener` and `remove EventListener`.
In conclusion, the problem here was not the DOM is not rendered with `render` method, but it needs some special handling for custom events.
Let me show you one example(the first one) and assign the rest to you.
```
test(
	'should emit the event after the delay',
	(done) => {
		const Component = DebounceDecorator(
			{debounce: 'onChange', delay: 100},
			function Base ({onChange}) {
				const ref = useRef(null);

				useEffect(() => {
					const {current} = ref;
					current.addEventListener('onChange', onChange);

					return () => {
						current.removeEventListener('onChange', onChange);
					};
				}, [onChange]);

				return <div ref={ref}>Test</div>;
			}
		);

		const spy = jest.fn();

		render(<Component onChange={spy} />);

		const elem = screen.getByText('Test');
		fireEvent(elem, createEvent('onChange', elem));

		expect(spy).not.toHaveBeenCalled();

		setTimeout(() => {
			expect(spy).toHaveBeenCalled();
			done();
		}, 150);
	}
);
```

Reply:
Thank you!